RUFAS/biophysical/manure/digester/anaerobic_digester.py

```diff
@@ -100,58 +75,42 @@ def process_manure(self, conditions: CurrentDayConditions, time: RufasTime) -> d
-                top_cover_volume=0.0,
+                captured_biogas_volume=0.0,
+                captured_methane_volume=0.0,
+                methane_leakage_volume=0.0,
                 simulation_day=time.simulation_day,
             )
             return {}
 
-        moisture_fraction = self._manure_in_digester.water / self._manure_in_digester.mass
-        specific_input_energy = self._calculate_specific_input_energy(
-            conditions.mean_air_temperature, moisture_fraction, self._temperature_set_point
-        )
-        heating_input_energy = (
-            specific_input_energy * self._manure_in_digester.volume * GeneralConstants.LITERS_TO_CUBIC_METERS
-        )
-
-        minimum_digester_volume = self._manure_in_digester.volume * self._hydraulic_retention_time
-        top_cover_volume = minimum_digester_volume * self._top_cover_volume_fraction
-
         self._manure_in_digester.ammoniacal_nitrogen = min(
             self._manure_in_digester.ammoniacal_nitrogen * TAN_INCREASE_FACTOR, self._manure_in_digester.nitrogen
         )
 
+        methane_density = 16.04 / (0.0821 * (self._temperature_set_point + 273.15))
+        carbon_dioxide_density = 44.01 / (0.0821 * (self._temperature_set_point + 273.15))
+
         generated_methane_volume = self._calculate_CSTR_methane_volume(self._manure_in_digester.total_volatile_solids)
-        generated_methane_mass = generated_methane_volume * METHANE_DENSITY
+        generated_methane_mass = generated_methane_volume * methane_density
         generated_carbon_dioxide_mass = (
-            generated_methane_volume * CARBON_DIOXIDE_TO_METHANE_RATIO * CARBON_DIOXIDE_DENSITY
-        )
+            generated_methane_volume * CARBON_DIOXIDE_TO_METHANE_RATIO) * carbon_dioxide_density
+        generated_carbon_dioxide_volume = generated_carbon_dioxide_mass / carbon_dioxide_density
+
+        total_biogas_volume = generated_methane_volume + generated_carbon_dioxide_volume
+        captured_biogas_volume = total_biogas_volume * (1 - self._biogas_leakage_fraction)
+
         total_volatile_solids_destruction = generated_methane_mass + generated_carbon_dioxide_mass
         self._manure_in_digester = self._destroy_volatile_solids(total_volatile_solids_destruction, time)
 
         self._manure_in_digester.volume -= total_volatile_solids_destruction / ManureConstants.SLURRY_MANURE_DENSITY
 
-        methane_leakage = self._calculate_methane_leakage(generated_methane_mass, self._methane_leakage_fraction)
-        captured_methane_mass = generated_methane_mass - methane_leakage
-        captured_methane_energy_content = self._calculate_methane_energy_content(captured_methane_mass)
-
-        evaporated_water = self._manure_in_digester.volume * self._evaporation_fraction
+        methane_leakage_volume = self._calculate_methane_leakage(generated_methane_mass, self._biogas_leakage_fraction)
+        captured_methane_volume = generated_methane_mass / methane_density - methane_leakage_volume
```

Comment:
```suggestion
        captured_methane_volume = generated_methane_volume - methane_leakage_volume
```
We already have a `generated_methane_volume` in line 92